### DISCUSSION TITLE
What to do if Metals keeps getting stuck compiling sub-projects?

### DISCUSSION THREAD

**Discussion by pepper-lees** [scalameta/nvim-metals#612] [Q&A]
Hey,

I've been fighting with this issue for a while. I've tried wiping and re-installing Metals, as well as running `:MetalsResetWorkspace` and `:MetalsRestartMetals` and `:MetalsCompileCancel`. I keep getting to a point where Metals attempts to recompile all my sources and then inevitably gets stuck at _some_ point in one of the sub-projects.

![image](https://github.com/scalameta/nvim-metals/assets/11487870/102cc18d-af26-46f5-8c49-e8d8ee05943e)

I don't know how to predict which project and which percentage it'll get stuck at, but it seems to always do this. What can I do to stop this from happening / what do I need to do if it does happen?

Thanks!

**Comment by pepper-lees** [comment]
Also, is there a way to prevent it from compiling everything, unless my code actually needs it? For example, I don't need to compile [hwacha](https://github.com/ucb-bar/hwacha/tree/bf799dc48293cb5017ed2ec22c5023de8d461184) or [midas](https://github.com/firesim/firesim/tree/08b565c8c69f125a1de292eb8efe03ee08408d2e/sim/midas) when I'm only importing from [rocketchip](https://github.com/chipsalliance/rocket-chip/tree/c563f74a54e60745969a5ad6b55a8207074f7509). All of the sub-projects are defined in the root build.sbt, but not all of them are being imported by the project files I am working on.

**Reply by ckipp01** [reply]
Hey @JL102, so regarding the getting stuck, at times the notifications when not using metals status aren't super accurate. I'd recommend to use metals status instead of the default which it seems you're doing. You can see more about this [here](https://github.com/scalameta/nvim-metals/blob/f41c14ae8500ceccb71e6695574b67881f0b5a93/doc/metals.txt#L101-L126).

Regarding the compilation of sub projects. I'm surprised by this. Could you provide me some steps to try and reproduce this?

**Reply by pepper-lees** [reply]
Hmm! *Maybe* I just got lucky with my current configuration, but after changing the status bar provider as you suggested, I completely stopped getting the behavior I described. Fingers crossed that it's permanent. If I get the same behavior again I'll get back to you! Thanks 😄

**Comment by Austinito** [comment]
I remember running into this very same issue. I was able to resolve it by using a different Java vendor from `SDKMAN!`
